29 token implementation

milenaf30 opened the pull request:
Se implementa la logica del login con token.

milenaf30 committed:
Se agrega configuracion de notificaciones a travis

PaulaSaffioti committed:
agrego soporte para POST y GET de users

PaulaSaffioti committed:
sudo primera version de Dockerfile

PaulaSaffioti committed:
subo primera version de tests de la api

PaulaSaffioti committed:
Merge branch 'ApplicationServer' of https://github.com/martinvol/tallerII into ApplicationServer

PaulaSaffioti committed:
corrijo readme

PaulaSaffioti committed:
actualizo Dockerfile. Ya es posible correr el server desde un contenedor

PaulaSaffioti committed:
agrego minima version requerida para evitar warning de docker

PaulaSaffioti committed:
script para correr tests

milenaf30 committed:
Refactor: se agregan comentarios al CMake para hacrlo más descriptivo

milenaf30 committed:
Refactor: se siguen agregando comentarios al CMake

milenaf30 committed:
Refactor: se borran elementos innecesarios en la configuracion del CMake

milenaf30 committed:
Se intentan agregar pruebas para chequear el funcionamiento de leveldn

milenaf30 committed:
libreria de leveldb para funcionar en travis

milenaf30 committed:
s cambia el nombre de la libreria de mongoose

milenaf30 committed:
se cambian los archivos de lugar

milenaf30 committed:
Se agregan .h a la definicion del ejecutable

milenaf30 committed:
Se sigue probando para que no pinchen las dependencias

milenaf30 committed:
se sigue probando sacando /lebeldb

milenaf30 committed:
Se siguen sacando leveldb/

milenaf30 committed:
Se modifica la ruta para que travis pueda encontrar los archivos

PaulaSaffioti committed:
Agrego posibilidad de generar un paquete .deb

PaulaSaffioti committed:
Vuelvo cambios para atras y lo saco del readme

PaulaSaffioti committed:
subo primera version de la documentacion

milenaf30 committed:
Se agrega el servicio de docker a travis

PaulaSaffioti committed:
pequeño cambio en documentacion

MartinCura committed:
min version al CMake anidado para que no llore

milenaf30 committed:
Se cambia el travis.yml para trabajar con docker. Primeras pruebas

milenaf30 committed:
Merge branch 'ApplicationServer' of https://github.com/martinvol/tallerII into ApplicationServer

milenaf30 committed:
Se agrega un update para ver si reconoce el comando build de docker

milenaf30 committed:
Se borran elementos de travis innecesarios ahora que se trabaja con docker

milenaf30 committed:
Corregi un path que estaba mal en travis.yml

milenaf30 committed:
Agregue un temporizador para esperar a que el servidor se levante en travis.yml

milenaf30 committed:
Se agrega un comentario descriptivo en el cmake

milenaf30 committed:
Se agrega un comentario descriptivo en el cmake

milenaf30 committed:
Se cambia el nombre de la libreria jsoncpp

milenaf30 committed:
Se agregan comentarios descriptivos en .travis.yml

milenaf30 committed:
Se agrega la opcion de correr los test desde cmake al dockerfile que crea la imagen

milenaf30 committed:
No reconoce make test. Habrá que investigar

milenaf30 committed:
Se sacan elementos innecesarios del cmake de leveldb para ahorrar tiempos de compilacion

PaulaSaffioti committed:
cambio manera en que se ejecuta el server para poder liberar memoria cuando finaliza

PaulaSaffioti committed:
Agrego primera version de log

PaulaSaffioti committed:
agrego include de Logger.h en main

milenaf30 committed:
Se cambia la ubicacion del DockerFile y se cambia su contenido para lograr que make test pueda ejecutarse

milenaf30 committed:
Se prueba si el ; al final aguarda a que el container se levante antes de hacer la conexion

milenaf30 committed:
Se cambia la ubicacion del DOckerfile en .travis

milenaf30 committed:
Se cambia la ubicacion del DOckerfile en .travis

milenaf30 committed:
Se agrega la libreria googletest

milenaf30 committed:
Se agrega gtest y se cambia el docker para que al menos se muestre la impresion del resultado del test

milenaf30 committed:
Se borra porque no acepta la carpeta

milenaf30 committed:
se vuelve a ingresar

milenaf30 committed:
Se agrega una carpeta donde guardar los tests

PaulaSaffioti committed:
refactor de logger

PaulaSaffioti committed:
agrego un lock para que varios threads no intenten escribir en el archivo al mismo tiempo

PaulaSaffioti committed:
uso libreria para armar json

PaulaSaffioti committed:
agrego content-length para solucionar error que habia en el primer request de get de users

PaulaSaffioti committed:
devuelvo int en vez de string

PaulaSaffioti committed:
agrego un timestamp a los logs

PaulaSaffioti committed:
agrego logs de error si no se puedo levantar el server

PaulaSaffioti committed:
saco metodo no usado

PaulaSaffioti committed:
Agrego objeto persona para encapsular los datos de las personas, y agrego algunos datos a la response del get de usuarios

PaulaSaffioti committed:
agrego work history

PaulaSaffioti committed:
agrego skills

PaulaSaffioti committed:
cambio fixme por algo más descriptivo

PaulaSaffioti committed:
Agrego otra persona

PaulaSaffioti committed:
mejor manejo de excepciones

PaulaSaffioti committed:
muevo logica al metodo

PaulaSaffioti committed:
using custom exception

PaulaSaffioti committed:
fix test

PaulaSaffioti committed:
renombro metodo

PaulaSaffioti committed:
cambio camelCase por snake_case

PaulaSaffioti committed:
arreglo tipos de datos

PaulaSaffioti committed:
arreglo tipos de datos

milenaf30 committed:
Refactor: saque codigo innecesario

milenaf30 committed:
Se agrega la opcion de eliminar el objeto de la basededatos

milenaf30 committed:
Se agregan los tests que deberan ser implementados para el ABM de usuarios

milenaf30 committed:
Se agregan pruebas de funcionalidad

milenaf30 committed:
Se agregan pruebas que a la vez son parte de la implementación del guardado de usuarios en la base de datos

milenaf30 committed:
Refactor: Se saca la creacion del usuario falso a un metodo

milenaf30 committed:
Se agrega a la clase persona la posibilidad de Serializarse y DesSerializarse con respecto a Json

milenaf30 committed:
Testing: Se agregan pruebas que corroboran la funcionalidad de la clase Person

milenaf30 committed:
Refactor: se extrae cierta funcionaliad a un metodo

milenaf30 committed:
Se agrega la opcion de deserializar

milenaf30 committed:
Tests: se agregan tests para comprobar el serializador de Personas

milenaf30 committed:
Se agrega la posibilidad de crear un objeto WorkHistory a partir de un json

milenaf30 committed:
Se agrega el serializador de Skill y se corrige una falta de return en WorkHstory

milenaf30 committed:
Refactor: se sacan elementos innecesarios en la clase Person

milenaf30 committed:
Prueba completa de person para generar json y obtener (se agrego Skill y WorkHistory)

milenaf30 committed:
Inicio de ABM de usuarios

PaulaSaffioti committed:
uso metodo de person para armar la response

PaulaSaffioti committed:
extraigo funcionalidad a un metodo

milenaf30 committed:
cambio en gitignore para que no suba configuracion de entorno de trabajo

milenaf30 committed:
Merge remote-tracking branch 'origin/ApplicationServer' into ApplicationServer

milenaf30 committed:
Se agrega una clase que maneja la creacion de la base de datos de manera de que solo exista una única instancia de la misma a lo largo de la aplicación.

milenaf30 committed:
Se saca una linea vieja que habia quedado

milenaf30 committed:
Se agregan tests para comprobar que la unica instancia de leveldb funcione y para comprobar que el id inicial esté bien seteado

milenaf30 committed:
Se borra una linea que habia quedado innecesaria

milenaf30 committed:
Primeros pasos en el guardado de usuarios.

PaulaSaffioti committed:
saco virtual

PaulaSaffioti committed:
agrego dos imagenes de perfil de prueba

PaulaSaffioti committed:
cambio en como se genera el string de headers

PaulaSaffioti committed:
refactor: muevo todos los handlers a una carpeta

PaulaSaffioti committed:
extraigo funcionalidad comun

PaulaSaffioti committed:
libero memoria

PaulaSaffioti committed:
agrego handler para las imagenes, por el momento sin funcionalidad

PaulaSaffioti committed:
muevo metodo a clase padre

PaulaSaffioti committed:
refactor: renombro metodo

PaulaSaffioti committed:
extraigo a una variable para mayor claridad

PaulaSaffioti committed:
agrego posibilidad de leer un archivo, con su correspondiente manejo de errores. Por el momento no devuelve el archivo.

PaulaSaffioti committed:
utilizo mg_send_http_chunk en vez de mg_printf_http_chunk para enviar respuesta al buffer de salida

PaulaSaffioti committed:
agrego posibilidad de enviar una imagen en la response

PaulaSaffioti committed:
Agrego set del id de las personas porque faltaba y por eso siempre devolvia cero

PaulaSaffioti committed:
saco atributo profile picture porque ya no es necesario

PaulaSaffioti committed:
Agrego tests de get de usuarios 1 y 2

PaulaSaffioti committed:
rearmado de la response si el body quedo invalido

milenaf30 committed:
getPersonByID

milenaf30 committed:
tests: getPersonByID

milenaf30 committed:
Se agrega un contador de id

milenaf30 committed:
Refactor: se mejoran estilos

milenaf30 committed:
Refactor: se extrae funcionalidad

milenaf30 committed:
Refactor: se extrae funcionalidad

milenaf30 committed:
Refactor: se cambian nombres de tests, para darle más sentido

milenaf30 committed:
Refactor: se cambia la excepcion UserNotFound para que muestre distintos tipos de mensaje.

milenaf30 committed:
Refactor: se agregan comentarios a los tests

PaulaSaffioti committed:
Muevo excepcion a la carpeta donde estan las demas, para mejor claridad

PaulaSaffioti committed:
Saco log redundante ya que existe otro que registra el mismo evento

PaulaSaffioti committed:
Agrego un end of line extra en el ultimo log para identificar facilmente diferentes corridas del server en el log file

PaulaSaffioti committed:
Agrego una carpeta para los logs y cambio para que se agregue un file distinto por mes

PaulaSaffioti committed:
agrego un break porque no tiene sentido seguir iterando el body si se sabe que es invalido

PaulaSaffioti committed:
Agrego log informativo si es necesario rearmar la response

PaulaSaffioti committed:
agrego referencia a "this" que faltaba

PaulaSaffioti committed:
Actualizacion de la documentacion

milenaf30 committed:
Refactor: se guarda el json si en algun momento se produce.

milenaf30 committed:
FixedError

milenaf30 committed:
GetPerson's Friends

milenaf30 committed:
DB Wrapper using Facade

milenaf30 committed:
DB Wrapper using Facade

milenaf30 committed:
Merge branch 'ApplicationServer' of https://github.com/martinvol/tallerII into ApplicationServer

PaulaSaffioti committed:
Logs en español

PaulaSaffioti committed:
Debug logs de los request y responses

PaulaSaffioti committed:
objeto para encapsular la información de Location

PaulaSaffioti committed:
agrego Location a Person

PaulaSaffioti committed:
test de location a traves de api

PaulaSaffioti committed:
inicializacion de Location en ambos constructores

PaulaSaffioti committed:
agrego Location al test de Person

PaulaSaffioti committed:
paso la url a DETELE y PUT requests

PaulaSaffioti committed:
refactor: elimino metodo y muevo logica a donde era llamado

PaulaSaffioti committed:
POST de users.

PaulaSaffioti committed:
elimino metodo que ya no se usa

PaulaSaffioti committed:
Agrego validación de que el email exista, sino lanza excepcion de que el request es incorrecto

PaulaSaffioti committed:
fix test

PaulaSaffioti committed:
pequeño refactor

PaulaSaffioti committed:
Agrego validación de que se envie la password

PaulaSaffioti committed:
algunos comentarios de fixes que hay que hacer

PaulaSaffioti committed:
saco tests hasta que la base de datos soporte crear y updetear un usuario correctamente

PaulaSaffioti committed:
intefraz para obtener all users

PaulaSaffioti committed:
extraigo funcionalidad a un metodo

PaulaSaffioti committed:
Agrego posibilidad de eliminar imagen de perfil

milenaf30 committed:
Merge remote-tracking branch 'origin/ApplicationServer' into ApplicationServer

# Conflicts:
#	ApplicationServer/src/api/handler/UsersHandler.cpp
#	ApplicationServer/src/person/Person.cpp
#	ApplicationServer/src/person/PersonManager.cpp
#	ApplicationServer/src/person/PersonManager.h

milenaf30 committed:
Fix: se saco el contenido basura que se mostraba inicialmente en la respuesta|

milenaf30 committed:
Conexion del get con la base de datos.
Se inicializa con user_1
cambios de int a long
generacion de id no secuencial

milenaf30 committed:
Implementacion  de PersonManager.getAllUsers

milenaf30 committed:
Se paso de vector<string> a vector<long>
Se hace un test que prueba que se devuelva correctamnete la lista

PaulaSaffioti committed:
pequeño cambio para que el get de las imagenes vuelva a funcionar

PaulaSaffioti committed:
no se envia content-type si no hay body

PaulaSaffioti committed:
get contacts de un user

PaulaSaffioti committed:
Agrego catch de excepciones no capturadas y devuelvo un 500 en ese caso

milenaf30 committed:
Se realizo el handler para el login
Se agrego la logica del login al PersonManager
Se armaron los tests que prueban su funcionamiento

milenaf30 committed:
Merge remote-tracking branch 'origin/ApplicationServer' into ApplicationServer

# Conflicts:
#	ApplicationServer/src/api/WebHandler.cpp

PaulaSaffioti committed:
fix class name

milenaf30 committed:
Fix: ERror de tipos

milenaf30 committed:
SessionManager
Se elimina variable interna de PersonManager innecesaria.
Tests de Session Manager iniciados

milenaf30 committed:
Merge remote-tracking branch 'origin/ApplicationServer' into ApplicationServer

milenaf30 committed:
SessionManager agregado al loginHandler
Se sacaron los fakeperson
Se generaron excepciones descriptivas

milenaf30 committed:
Se cambio el init para que tenga password

PaulaSaffioti committed:
excepcion custom en vez de generica

PaulaSaffioti committed:
Inicializacion de la base de datos cuando carga el server

PaulaSaffioti committed:
delete de personManager que faltaba

PaulaSaffioti committed:
refactor para hacer un correcto delete del personManager

PaulaSaffioti committed:
saco tests de POST de users hasta que este bien definida esa interfaz

PaulaSaffioti committed:
vuelvo a agregar linea comentada

PaulaSaffioti committed:
actualizacion de getContacts sin fake users

PaulaSaffioti committed:
seteo correcto de user id y user email en excepcion

PaulaSaffioti committed:
cambio para que si un usuario ya existe no lo cree, pero continue con la carga de los demas

PaulaSaffioti committed:
hasta que no este definida la interfez de POST de users, devolvera Not Implemented

PaulaSaffioti committed:
fix typo

PaulaSaffioti committed:
metadata con cantidad total de usuarios en GET de allusers

PaulaSaffioti committed:
delete de usuarios

PaulaSaffioti committed:
saco test porque no es unitario

PaulaSaffioti committed:
extraccion de delete a un metodo

PaulaSaffioti committed:
metodo para updetear una persona

PaulaSaffioti committed:
primera version de update de people

PaulaSaffioti committed:
saco test de delete de users porque si hay retry de tests fallan

PaulaSaffioti committed:
fix para que si un dato no se envia, quede vacio

PaulaSaffioti committed:
catch de excepciones para que devuelva la response correcta y no muera el server

PaulaSaffioti committed:
uso de "email" en vez de "mail", porque asi esta definido en la API

PaulaSaffioti committed:
refactor para evitar perdida de memoria

PaulaSaffioti committed:
return que faltaba

PaulaSaffioti committed:
creacion de usuarios

PaulaSaffioti committed:
fix de algunos issues de updates de people

PaulaSaffioti committed:
remuevo logica para que no devuelva password

PaulaSaffioti committed:
correcto borrado de elementos de vector

PaulaSaffioti committed:
referencia a this que faltaba

milenaf30 committed:
Se corrigio el login que fallaba porque se habia borrado el campo del password,
   Se corrigio para que si el login falla, mande un mensaje descriptivo.
   Se corrigio un espacio que faltaba en el mensaje de error de UserNotFound
   Se corrigio para que los mails del DBBuilder comiencen en minúscula

milenaf30 committed:
removeMember solo agrega "" al password para que no se guarde en la base

milenaf30 committed:
Se modifico la creaciń del id unico para evitar que sea de valor negativo.

milenaf30 committed:
Agregue para que el resultado del login aparezca en el logger

milenaf30 committed:
Se agrega el id del usuario en la respuesta del login

PaulaSaffioti committed:
WIP: PUT de una imagen

PaulaSaffioti committed:
fix test porque se cambio de mayuscula a minuscula en la inicializacion de la DB

PaulaSaffioti committed:
WIP: agregar y aceptar solicitudes de amistad

PaulaSaffioti committed:
eliminar solicitudes y contactos

PaulaSaffioti committed:
ignora extra data que viene en el body para guardar la imagen

PaulaSaffioti committed:
uso de long en vez de int

PaulaSaffioti committed:
refactor: nombre de metodo mas acorde

PaulaSaffioti committed:
invalid request si la action es invalida

PaulaSaffioti committed:
test de contacts

PaulaSaffioti committed:
set de tests individuales de cada interfaz

PaulaSaffioti committed:
WIP: servicio de recomendaciones

PaulaSaffioti committed:
servicio de recomendaciones terminado

PaulaSaffioti committed:
devuelve array vacio en vez de null si no hay recomendaciones

PaulaSaffioti committed:
WIP: servicio de mensajes

PaulaSaffioti committed:
Servicio de mensajes terminado

PaulaSaffioti committed:
fix para que devuelva como maximo 10 mensajes y cuente desde el mas nuevo hacia el mas viejo

PaulaSaffioti committed:
fix para no tener que enviar body en los DELETE requests

PaulaSaffioti committed:
from como parameter

PaulaSaffioti committed:
devuelve todos los mensajes (no en orden aun)

PaulaSaffioti committed:
devuelve todos los mensajes en orden

milenaf30 committed:
Se agrega al CMake la dependencia faltante a RelationManager.*

milenaf30 committed:
Se agregan métodos correspondientes para mostrar el mensaje 401 cuando se intenta iniciar sesión y no se puedo autenticar

milenaf30 committed:
#29 Cambios de sintaxis en el mensaje de error.

milenaf30 committed:
#29 Se agrego la logica de chequear tokens para metodos privados

milenaf30 committed:
#29 Se cambia de lugar la logica de inicio de sesión.

milenaf30 committed:
#29 Se agrega metodo de status code 401

milenaf30 committed:
#29 Se agrega un atributo por cada accion http

milenaf30 committed:
close #29 Se agrego excepcion bad request cuando no tiene el encabezado correspondiente en el header

martinvol commented:
Mañana puedo arreglar el conflicto y que queden los dos travis andando.

milenaf30 committed:
se cambia la pass

MartinCura commented:
Testeado con el cliente y parece andar todo bien.
Visto bueno para mergear.
Si cambiás alguna de las cosas que te dije por WhatsApp o el Docs (ahora no me acuerdo jaja) avisame!

MartinCura commented:
Cuidado que estarías mergeando al master según veo, en vez de al ApplicationServer (con cual ni tendrías problemas de travis que yo sepa).

milenaf30 committed:
Se agrega documentación a los métodos.
Se cambian algunos valores de los json guardados para cumplir cierta estética

milenaf30 committed:
Se quita codigo innecesario
Se inicializa una variable que no estaba inicializada anteriormente y se devolvia

milenaf30 committed:
Se agrega clase que guarda información de la sesion
Se agrega clase que controla los permisos de las acciones
Se actualizan tests
Se cambio el tipo de retorno de algunos métodos.

milenaf30 committed:
Se agregan permisos para eliminarun usuario

milenaf30 committed:
Se agregan permisos para mandar y leer mensajes

milenaf30 committed:
Se agregan permisos para editar y borrar imagenes de perfil

milenaf30 committed:
Se mueve control para evitar perdidas de memoria

milenaf30 committed:
Se agregan permisos para editar y borrar imagenes de perfil

milenaf30 committed:
Se agregan permisos para editar y borrar imagenes de perfil

milenaf30 committed:
Se agregan permisos para enviar y borrar recomendaciones
Se agrega el metodo de getAllUsers publico

milenaf30 committed:
Se agrega el status code correspondiente para indicar que el usuario no tiene permisos.

milenaf30 committed:
Cambio status code devuelto por el login cuando el usuario que solicita el inicio de sesión no existe en el sistema.

milenaf30 committed:
fix: me habia olvidado un return

MartinCura commented:
@milenaf30 Se cierra este, ¿no?
